sgv2-service-common/src/main/java/io/stargate/sgv2/common/metrics/SchemaCacheStatsRecorder.java

```diff
@@ -0,0 +1,85 @@
+package io.stargate.sgv2.common.metrics;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.github.benmanes.caffeine.cache.stats.CacheStats;
+import com.github.benmanes.caffeine.cache.stats.StatsCounter;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation inspired by <a
+ * href="https://metrics.dropwizard.io/4.2.0/manual/caffeine.html">Dropwizart-metrics for
+ * Caffeine</a> but modified for out needs.
+ */
+public class SchemaCacheStatsRecorder implements StatsCounter {
```

Comment:
Caffeine cache stats are supported out of the box with the Micrometer: https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/CaffeineCacheMetrics.java

Reply:
Supporting Caffeine stats is either available via Dropwizard Metrics Caffeine plugin -- single-class -- as well.

But what I can do first, I think, is just to drop Cache instrumentation from this PR: I can see that cache hit rate is high so no need to verify. And we can tackle metrics side separately.

It does still leave the question of other metrics but at least removes one distraction.

Reply:
Removed this class & cache instrumentation; separate issue.